cmd/backup-manager/app/backup/backup.go

```diff
@@ -134,3 +134,124 @@ func constructOptions(backup *v1alpha1.Backup) ([]string, error) {
+func (bo *Options) startlogBackupExec(ctx context.Context, backup *v1alpha1.Backup) error {
+	specificArgs := []string{
+		"log",
+		"start",
+		fmt.Sprintf("--task-name=%s", backup.Name),
+	}
+	if backup.Spec.CommitTs != "" {
+		specificArgs = append(specificArgs, fmt.Sprintf("--start-ts=%s", backup.Spec.CommitTs))
+	}
+	fullArgs, err := bo.logBackupCommandTemplate(backup, specificArgs)
+	if err != nil {
+		return err
+	}
+	return bo.brCommandRun(ctx, fullArgs)
+}
+
+// stoplogBackupExec generates br args about log backup stop and runs br binary to do the real backup work.
+func (bo *Options) stoplogBackupExec(ctx context.Context, backup *v1alpha1.Backup) error {
+	specificArgs := []string{
+		"log",
+		"stop",
+		fmt.Sprintf("--task-name=%s", backup.Name),
+	}
+	fullArgs, err := bo.logBackupCommandTemplate(backup, specificArgs)
+	if err != nil {
+		return err
+	}
+	return bo.brCommandRun(ctx, fullArgs)
+}
+
+// truncatelogBackupExec generates br args about log backup truncate and runs br binary to do the real backup work.
+func (bo *Options) truncatelogBackupExec(ctx context.Context, backup *v1alpha1.Backup) error {
+	specificArgs := []string{
+		"log",
+		"truncate",
+		fmt.Sprintf("--until=%s", backup.Spec.LogTruncateUntil),
+	}
+	fullArgs, err := bo.logBackupCommandTemplate(backup, specificArgs)
+	if err != nil {
+		return err
+	}
+	return bo.brCommandRun(ctx, fullArgs)
+}
+
+// logBackupCommandTemplate is the template to generate br args.
+func (bo *Options) logBackupCommandTemplate(backup *v1alpha1.Backup, specificArgs []string) ([]string, error) {
+	if len(specificArgs) == 0 {
+		return nil, fmt.Errorf("log backup command is invalid, Args: %v", specificArgs)
+	}
+
+	clusterNamespace := backup.Spec.BR.ClusterNamespace
+	if backup.Spec.BR.ClusterNamespace == "" {
+		clusterNamespace = backup.Namespace
+	}
+	args := make([]string, 0)
+	args = append(args, fmt.Sprintf("--pd=%s-pd.%s:2379", backup.Spec.BR.Cluster, clusterNamespace))
+	if bo.TLSCluster {
+		args = append(args, fmt.Sprintf("--ca=%s", path.Join(util.ClusterClientTLSPath, corev1.ServiceAccountRootCAKey)))
+		args = append(args, fmt.Sprintf("--cert=%s", path.Join(util.ClusterClientTLSPath, corev1.TLSCertKey)))
+		args = append(args, fmt.Sprintf("--key=%s", path.Join(util.ClusterClientTLSPath, corev1.TLSPrivateKeyKey)))
+	}
+	// `options` in spec are put to the last because we want them to have higher priority than generated arguments
+	dataArgs, err := constructOptions(backup)
+	if err != nil {
+		return nil, err
+	}
+	args = append(args, dataArgs...)
+
+	fullArgs := append(specificArgs, args...)
+	return fullArgs, nil
+}
+
+// brCommandRun run br binary to do backup work
+func (bo *Options) brCommandRun(ctx context.Context, fullArgs []string) error {
+	if len(fullArgs) == 0 {
+		return fmt.Errorf("command is invalid, fullArgs: %v", fullArgs)
+	}
+	klog.Infof("Running br command with args: %v", fullArgs)
+	bin := path.Join(util.BRBinPath, "br")
```

Comment:
You'd better use `filepath` package. The `path` package is not compatible with all file systems.